lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -276,7 +276,13 @@ def main():
     # find the group if present
     group = None
     groups = {}
-    for curGroup in ec2.get_all_security_groups():
+
+    try:
+        security_groups = ec2.get_all_security_groups()
+    except Exception as e:
```

Comment:
We do have guidelines on boto3 exception handling
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2

`botocore.exceptions.ClientError` covers most needs, and is a little better than just catching `Exception`